components/net/http_loader.rs

```diff
@@ -1100,17 +1102,20 @@ pub async fn http_redirect_fetch(
 /// [HTTP network or cache fetch](https://fetch.spec.whatwg.org#http-network-or-cache-fetch)
 #[async_recursion]
 async fn http_network_or_cache_fetch(
-    request: &mut Request,
+    fetch_params: &mut FetchParams,
     authentication_fetch_flag: bool,
     cors_flag: bool,
     done_chan: &mut DoneChannel,
     context: &FetchContext,
 ) -> Response {
     // Step 1. Let request be fetchParams’s request.
-    // NOTE: We get request as an argument (Fetchparams are not implemented, see #33616)
+    let request = &mut fetch_params.request;
+
+    // Step 2. Let httpFetchParams be null.
+    let http_fetch_params: &mut FetchParams;
+    let mut fetch_params_copy: FetchParams;
```

Comment:
needed to make a variable to hold a clone of fetch params since in step 8.2 it's required to clone it